spectrum/input.py

```diff
@@ -108,37 +108,53 @@ def create_blog_article(self, title, text='Lorem ipsum', image=None):
         # TODO: return id and/or node id
 
     def create_article_fragment(self, id, image):
-        create_url = "%s/admin/structure/article_fragment/add" % self._host
-        create_page = self._browser.get(create_url)
-        form = mechanicalsoup.Form(create_page.soup.form)
-        form.input({'name[0][value]': id})
-        form.attach({'files[image_0]': image})
+        filtered_content_url = "%s/admin/content?status=All&type=article&title=%s" % (self._host, id)
+        filtered_content_page = self._browser.get(filtered_content_url)
+
+        try:
+            view_url = "%s%s" % (self._host, filtered_content_page.soup.find('td', 'views-field-title').find('a', href=True, text=id).get('href'))
+            edit_url = "%s%s" % (self._host, filtered_content_page.soup.find('td', 'views-field-operations').find('li', 'edit').find('a', href=True, text='Edit').get('href'))
+        except (AttributeError, TypeError):
+            raise AssertionError('View and edit link not found for article: %s' % id)
+
         LOGGER.info(
-            "Submitting thumbnail %s",
-            image,
+            "Access edit form",
             extra={'id': id}
         )
-        self._choose_submit(form, 'image_0_upload_button', value='Upload')
-        response = self._browser.submit(form, create_page.url)
-        form = mechanicalsoup.Form(response.soup.form)
 
+        edit_page = self._browser.get(edit_url)
+
+        form = mechanicalsoup.Form(edit_page.soup.form)
 
-        form.attach({'files[banner_image_0]': image})
+        if edit_page.soup.find('input', {'name': 'field_image_0_remove_button'}):
+            self._choose_submit(form, 'field_image_0_remove_button', value='Remove')
+            response = self._browser.submit(form, edit_page.url)
+            form = mechanicalsoup.Form(response.soup.form)
+            LOGGER.info(
+                "Removing existing thumbnail",
+                extra={'id': id}
+            )
 
+        form.attach({'files[field_image_0]': image})
         LOGGER.info(
-            "Submitting banner %s",
+            "Submitting thumbnail %s",
             image,
             extra={'id': id}
         )
-        self._choose_submit(form, 'banner_image_0_upload_button', value='Upload')
-        response = self._browser.submit(form, create_page.url)
+        self._choose_submit(form, 'field_image_0_upload_button', value='Upload')
+        response = self._browser.submit(form, edit_page.url)
         form = mechanicalsoup.Form(response.soup.form)
+
         LOGGER.info(
             "Saving form",
             extra={'id': id}
         )
-        response = self._browser.submit(form, create_page.url, data={'op': 'Save'})
-        img = response.soup.select_one(".field--name-banner-image img")
+
+        # Button text will be 'Save and keep published' or 'Save and keep unpublished'
+        button_text = edit_page.soup.find('div', {'id': 'edit-actions'}).find('input', 'form-submit').get('value')
```

Comment:
I think we should assert that we find only one input element here, not sure Beautiful Soup enforces it on `find()` but I'll check later (reminder for myself)